### DISCUSSION TITLE
I don't understand how "extends" work with a very basic example.

### DISCUSSION THREAD

**Discussion by hlassiege** [nuxt/nuxt#26005] [Questions]
Hi,

I'm working on an opensource project : https://github.com/hlassiege/bloggr

I wanted to try the **extends** configuration to create a theming system.

I first tried to create a project with almost nothing into it : 

https://github.com/hlassiege/bloggr-demo

And  there is one extends statement in the nuxt.config.ts file : 

```
    extends: ["github:hlassiege/bloggr"],
```

However I have some problem because in bloggr, I use dynamic import

```
const footerComponent = defineAsyncComponent(
    () => import(`../components/themes/${config.public.theme}/Footer.vue`),
);
```
And the resolution "../components" seems to fail when the file is in bloggr-demo (the one that extends bloggr)
```
 "Unknown variable dynamic import: ../components/themes/mistral/Header.vue",
```

I'm not sure if I understood clearly the documentation but maybe it's not possible 

https://nuxt.com/docs/guide/going-further/layers#relative-paths-and-aliases

> When importing using aliases (such as ~/ and @/) in a layer components and composables, note that aliases are resolved relative to the user's project paths. As a workaround, you can use relative paths to import them. We are working on a better solution for named layer aliases.

Wdyt ?

**Comment by AloisSeckar** [comment]
This issue looks pretty much as problem with [Relative Paths and Aliases](https://nuxt.com/docs/guide/going-further/layers#relative-paths-and-aliases).

Nuxt does not extend base layer's `/assets` by default, so you need to give it a hint where to find the files.

**Reply by hlassiege** [reply]
Indeed, there is some problem in the name resolution

I removed the include for the assets and I have now an error on this line : 
```
const headerComponent = defineAsyncComponent(
    () => import(`../components/themes/${config.public.theme}/Header.vue`),
);
```
=>
```
"Unknown variable dynamic import: ../components/themes/epoxia/Header.vue",
```

I tried ../components, ~/components
nothing works

**Reply by AloisSeckar** [reply]
The relative path doesn't look quite right, if you're referencing components from a remote package. If I understand the concept of layers correctly, they give you opportunity of using additional components, but they are not virtually copied into your project's components directory.

The mentioned component should be reachable as `ThemesEpoxiaHeader`, for example via resolveComponent function:
```
resolveComponent('ThemesEpoxiaHeader')
```
or as an import
```
import { ThemesEpoxiaHeader } from '#components'
```

There is a catch though. As [stated in the docs](https://nuxt.com/docs/guide/directory-structure/components#dynamic-components), it is not possible to reference really "dynamic" value like in vanilla Vue, but you need to specify the concrete `import` or pass plain string into `resolveComponent`. How many different options do you have? For a few options it is doable, but if you have like 10 diferent components and 5 possible themes, it would be a mess...

Also not sure how to combine it with async component. Do you need to use it as an async component?

**Reply by hlassiege** [reply]
yes, the goal is to create a theming system, that anyone can use to create their own theme. 

For the moment I use an alternative, I copy an archive directly into the source code. It's messy but it works
In the future I wonder how I can improve this to split the theme from the rest

(I'm thinking out loud, not waiting for an answer :) I'll continue to think about it)

**Reply by hlassiege** [reply]
so, the only remaining problem is indeed related to dynamic import
Everything else works perfectly but fails if the component is in bloggr-demo (the module that extends the other one)

const headerComponent = defineAsyncComponent(
    () => import(`../components/themes/${config.public.theme}/Header.vue`),
);

I've tried different syntax  #components,  ~components, ../components etc..

```
const headerComponent = resolveComponent(
    `~/components/themes/${config.public.theme}/Header.vue`,
);

const headerComponent = defineAsyncComponent(
    () => import(`#components/themes/${config.public.theme}/Header.vue`),
);
```

No success so far. I'm pretty sure I'm quite close. I'll continue to investigate

**Comment by hlassiege** [comment]
I found it :)

I read this discussion in #14036 and now I'm using : 

```
function capitalizeFirstLetter(str: string): string {
    return str.charAt(0).toUpperCase() + str.slice(1);
}

const theme = capitalizeFirstLetter(config.public.theme);

const headerComponent = resolveComponent(`Themes${theme}Header`);
const footerComponent = resolveComponent(`Themes${theme}Footer`);
const postComponent = resolveComponent(`Themes${theme}Post`);
```

And for this solution to work, I needed to add .global.vue for every components that could be dynamic
And it works !
